docker/docker_client.go

```diff
@@ -212,37 +210,46 @@ func hasFile(files []os.FileInfo, name string) bool {
 	return false
 }
 
-// newDockerClient returns a new dockerClient instance for refHostname (a host a specified in the Docker image reference, not canonicalized to dockerRegistry)
+// newDockerClientFromRef returns a new dockerClient instance for refHostname (a host a specified in the Docker image reference, not canonicalized to dockerRegistry)
 // “write” specifies whether the client will be used for "write" access (in particular passed to lookaside.go:toplevelFromSection)
-func newDockerClient(ctx *types.SystemContext, ref dockerReference, write bool, actions string) (*dockerClient, error) {
+func newDockerClientFromRef(ctx *types.SystemContext, ref dockerReference, write bool, actions string) (*dockerClient, error) {
 	registry := reference.Domain(ref.ref)
-	if registry == dockerHostname {
-		registry = dockerRegistry
-	}
 	username, password, err := getAuth(ctx, reference.Domain(ref.ref))
 	if err != nil {
+		return nil, errors.Wrapf(err, "error getting username and password")
+	}
+	sigBase, err := configuredSignatureStorageBase(ctx, ref, write)
+	if err != nil {
 		return nil, err
 	}
+	hostName := reference.Domain(ref.ref)
+	remoteName := reference.Path(ref.ref)
+
+	return newDockerClientWithDetails(ctx, registry, username, password, actions, sigBase, hostName, remoteName)
+}
+
+// newDockerClientWithDetails returns a new dockerClient instance for the given parameters
+func newDockerClientWithDetails(ctx *types.SystemContext, registry, username, password, actions string, sigBase signatureStorageBase, hostName, remoteName string) (*dockerClient, error) {
+	if registry == dockerHostname {
+		registry = dockerRegistry
+	}
 	tr := newTransport()
 	tr.TLSClientConfig = serverDefault()
+	client := &http.Client{Transport: tr}
```

Comment:
(Absolutely non-blocking: `client` is unused throughout the function; initializing it later, or just inlining `client: &http.Client{…}` in the set initializer would mean readers have less to track.)

Reply:
fixed